src/game_battlealgorithm.cpp

```diff
@@ -736,6 +740,25 @@ int Game_BattleAlgorithm::Normal::GetCBAMovement() const {
 	return lcf::rpg::BattlerAnimationItemSkill::Movement_none;
 }
 
+const lcf::rpg::BattlerAnimationItemSkill* Game_BattleAlgorithm::Normal::GetWeaponAnimationData() const {
+	const auto weapon = GetWeapon();
+	auto* source = GetSource();
+	if (source->GetType() == Game_Battler::Type_Ally) {
+		auto* ally = static_cast<Game_Actor*>(source);
+		auto weapons = ally->GetWeapons(weapon);
+		auto* item = weapons[0];
```

Comment:
Again, need to support dual wield here

Reply:
Please see my answer to your first comment. Sadly I need guidance to make this work the RPG_RT way.

Reply:
My bad, I forgot how my own code works. I believe this should work fine. `weapons[0]` should be the primary or second weapon based on which weapon is currently active from `GetWeapon()`